interface.go

```diff
@@ -89,20 +90,46 @@ type JobStatusInfo struct {
 
 // JobTimeInfo stores timing information for a job and is used by both
 // the Runner and Job implementations to track how long jobs take to
-// execute. Additionally, the Queue implementations __may__ use this
-// data to delay execution of a job when WaitUntil refers to a time
-// in the future.
+// execute.
+//
+// Additionally, the Queue implementations __may__ use WaitUntil to
+// defer the execution of a job, until WaitUntil refers to a time in
+// the past.
+//
+// If the deadline is specified, and the queue
+// implementation supports it, the queue may drop the job if the
+// deadline is in the past when the job would be dispatched.
 type JobTimeInfo struct {
-	Created   time.Time     `bson:"created,omitempty" json:"created,omitempty" yaml:"created,omitempty"`
-	Start     time.Time     `bson:"start,omitempty" json:"start,omitempty" yaml:"start,omitempty"`
-	End       time.Time     `bson:"end,omitempty" json:"end,omitempty" yaml:"end,omitempty"`
-	WaitUntil time.Time     `bson:"wait_until" json:"wait_until,omitempty" yaml:"wait_until,omitempty"`
-	MaxTime   time.Duration `bson:"max_time" json:"max_time,omitempty" yaml:"max_time,omitempty"`
+	Created    time.Time     `bson:"created,omitempty" json:"created,omitempty" yaml:"created,omitempty"`
+	Start      time.Time     `bson:"start,omitempty" json:"start,omitempty" yaml:"start,omitempty"`
+	End        time.Time     `bson:"end,omitempty" json:"end,omitempty" yaml:"end,omitempty"`
+	WaitUntil  time.Time     `bson:"wait_until" json:"wait_until,omitempty" yaml:"wait_until,omitempty"`
+	DispatchBy time.Time     `bson:"dispatch_by" json:"dispatch_by,omitempty" yaml:"dispatch_by,omitempty"`
+	MaxTime    time.Duration `bson:"max_time" json:"max_time,omitempty" yaml:"max_time,omitempty"`
 }
 
 // Duration is a convenience function to return a duration for a job.
 func (j JobTimeInfo) Duration() time.Duration { return j.End.Sub(j.Start) }
 
+// IsStale determines if the job is too old to be dispatched, and if
+// so, queues may remove or drop the job entirely.
+func (j JobTimeInfo) IsStale() bool { return !j.DispatchBy.After(time.Now()) }
```

Comment:
This doesn't work with zero time. The beginning of time is not after now and so the zero value is stale, which will cause unset values to be removed or dropped. I'm not sure there's anything you can do except deliberately check zero time.